pkg/frontend/openshiftcluster_preflightvalidation.go

```diff
@@ -88,23 +101,34 @@ func (f *frontend) _preflightValidation(ctx context.Context, log *logrus.Entry,
 		}
 	}
 
-	converter.ToInternal(ext, oc)
-	if err := staticValidator.Static(ext, nil, f.env.Location(), f.env.Domain(), f.env.FeatureIsSet(env.FeatureRequireD2sV3Workers), resourceID); err != nil {
-		return api.ValidationResult{
-			Status: api.ValidationStatusFailed,
-			Error: &api.ManagementErrorWithDetails{
-				Message: to.StringPtr(err.Error()),
-			},
+	if isCreate {
+		log.Print("preflight static create validation")
+		if err = staticValidator.Static(ext, nil, f.env.Location(), f.env.Domain(), f.env.FeatureIsSet(env.FeatureRequireD2sV3Workers), resourceID); err != nil {
+			return api.ValidationResult{
+				Status: api.ValidationStatusFailed,
+				Error: &api.ManagementErrorWithDetails{
+					Message: to.StringPtr(err.Error()),
+				},
+			}
 		}
-	}
-
-	if err := f.validateInstallVersion(ctx, oc); err != nil {
-		return api.ValidationResult{
-			Status: api.ValidationStatusFailed,
-			Error: &api.ManagementErrorWithDetails{
-				Code:    to.StringPtr(api.CloudErrorCodeInvalidParameter),
-				Message: to.StringPtr(err.Error()),
-			},
+		if err = f.validateInstallVersion(ctx, oc); err != nil {
+			return api.ValidationResult{
+				Status: api.ValidationStatusFailed,
+				Error: &api.ManagementErrorWithDetails{
+					Code:    to.StringPtr(api.CloudErrorCodeInvalidParameter),
+					Message: to.StringPtr(err.Error()),
+				},
+			}
+		}
+	} else {
+		log.Print("preflight static update validation")
```

Comment:
I noticed the story indicates to add update and delete but here it looks like this only covers updates?

Reply:
Correct, I after looking into preflight we saw that delete isn't a path thats going to be used by ARM and so we won't be adding that. Ill adjust the ticket.